### ISSUE TITLE
how to pretrain wav2vec2 on my own audio using an existing model as a base?/

### ISSUE THREAD

**Issue by tensorfoo** [pytorch/fairseq#3300]
i'm calling pretraining using 
`fairseq-hydra-train --run --config-name myconfig  --config-dir  examples/wav2vec/config/pretraining`

which ends up getting stuck on the same loss of 6.658. It never improves on that no matter what I've tried. 

Someone in another issue suggested using one of the base models for pretraining. I haven't come across where in the docs it tells you how to do that? Any help appreciated.

**Comment by jubick1337** [comment]
Hi @tensorfoo!
You can change it in your config.yaml by setting ```restore_file``` in checkpoint atribute. 
If checkpoint stored outside of ```./checkpoints``` (which is default dir for checkpoints) you should also set ```save_dir``` to proper dir.

**Comment by tensorfoo** [comment]
thank you @jubick1337 - I appreciate it! Turned out I couldn't use that method anyway, it says the model parameters are different. Unless I have the same HW configuration as that which was used to train the model, I am probably out of luck..

**Comment by olafthiele** [comment]
Try @mailong25 [repo ](https://github.com/mailong25/self-supervised-speech-recognition)for finetuning. Pretty good instructions to get you going. As for the parameters, if you don't share information on them or what you are doing, it is hard to give any advice.

**Comment by tensorfoo** [comment]
> Try @mailong25 [repo ](https://github.com/mailong25/self-supervised-speech-recognition)for finetuning. Pretty good instructions to get you going. As for the parameters, if you don't share information on them or what you are doing, it is hard to give any advice.

Hi olafthiele. The only changes i made to the `wav2vec2_base_librispeech.yaml` base file is adding a batch of 2 (because I have a 1080ti with 11gb of vram) which prevents OOM. And I also turned FP16 True to False and set distributed_world_size from 64 to 1. 

I will also try your suggestion, I really appreciate the help!

**Comment by olafthiele** [comment]
@tensorfoo 
1. Get a small set of 1 hour with audio clips of roughly 10-20 secs to work with finetuning and the same parameters as the repo mentioned above. Just to get stuff going.
2. Then start to tweak and change stuff to work with all of your material. If you change too many things at once it will be hard to debug it. Looks like you are mixing methods currently. Don't. Either go with the mailong repo or use fairseqs scripts. But mailong has a working setup and as long yours isn't, I would stick with a working setup.

**Comment by tensorfoo** [comment]
Hi @olafthiele, I was able to steal what mailong was doing right to modify the base script, the changes I added to my config: 

<   restore_file: /tmp/wav2vec_small.pt
<   reset_optimizer: true
<   reset_dataloader: true
<   reset_meters: true 

That seems to have done the trick. I am still unsure what successful training will look like. On my dataset the best loss i managed was 2.5ish, which came early in training and subsequent epochs just went further and further away, with losses growing on training and validation both, eventually getting stuck at the same place I  got when training from scratch. 

I am now doing another experiment with a public dataset, https://www.openslr.org/25/, 

the test set from the broadcast news specifically: https://www.openslr.org/resources/25/data_broadcastnews_sw.tar.bz2

so far validation hasn't broken 2.0 and i think it might not get there over time. I was expecting overfitting would mean loss would go down to zero on a small dataset but that hasn't been my experience? 

Could you please give me some intuition on what to expect when training is going well?

**Comment by olafthiele** [comment]
You don't give much information (e.g. what is a small dataset, 100h?). So it is hard to give any advice. You are finetuning for an African language from English. This can work to a degree and a loss of 2 doesn't sound too bad. BUT you need a golden testset for yourself to check checkpoints. Losses won't give you much indication for that sort of task. But if your testset wer still improves, keep on training ...
